srv/salt/ceph/startup/default.sls

```diff
@@ -0,0 +1,52 @@
+
+
+Starting mon:
+  salt.state:
+    - sls: ceph.start.mon
+    - tgt: "I@roles:mon and I@cluster:ceph"
+    - tgt_type: compound
+
+Starting mgr:
+  salt.state:
+    - sls: ceph.start.mgr
+    - tgt: "I@roles:mgr and I@cluster:ceph"
+    - tgt_type: compound
+
+Starting storage:
+  salt.state:
+    - sls: ceph.start.storage
+    - tgt: "I@roles:storage and I@cluster:ceph"
+    - tgt_type: compound
+
+{% if salt.saltutil.runner('select.minions', cluster='ceph', roles='igw') %}
+Starting iscsi:
+  salt.state:
+    - sls: ceph.start.igw
+    - tgt: "I@roles:igw and I@cluster:ceph"
+    - tgt_type: compound
+{% endif %}
+
+{% if salt.saltutil.runner('select.minions', cluster='ceph', roles='mds') %}
```

Comment:
I just realised something ... This (and other 'select.minions' blocks) shouldn't really be necessary, because if there's no mds roles, the tgt line below won't match any hosts. Or is there some other effect I'm forgetting?

Reply:
Salt will complain about "target matched no minions" and also count it as a failure.

Reply:
Ah, of course. Thanks.